src_c/freetype/ft_render.c

```diff
@@ -655,9 +655,12 @@ _PGFT_Render_Array(FreeTypeInstance *ft, pgFontObject *fontobj,
     }
     if (_validate_view_format(view_p->format)) {
         static const char fmt[] = "Unsupported array item format '%.*s'";
-        char msg[100 + sizeof(fmt)];
 
-        sprintf(msg, fmt, (int)(sizeof(msg) - sizeof(fmt)), view_p->format);
+        /* set a default error message incase PyOS_snprintf fails */
+        char msg[100 + sizeof(fmt)] = "Unsupported array item format";
+
+        PyOS_snprintf(msg, sizeof(msg), fmt, (int)(sizeof(msg) - sizeof(fmt)),
+                      view_p->format);
```

Comment:
I believe this one could use PyUnicode_FromFormat instead, which seems preferable. `view_p` is a Py_buffer, the `format` member is a null terminated char array. https://docs.python.org/3/c-api/buffer.html#c.Py_buffer.format

Reply:
Here `msg` is a C string, that function is for python unicode objects. Yeah that API can be used, and then another function to convert the python string to a C string would need calling, which is more code, and not as performant as `PyOS_snprintf`

Reply:
Well can't PyUnicode_Format or PyErr_Format take c strings?

Reply:
`PyErr_Format` seems to do the trick here, so I've implemented that, thanks!